src/test/java/com/tabnine/binary/BootstrapperSupportTests.java

```diff
@@ -54,6 +55,15 @@ public void testWhenBootstrapperVersionExistsItWillUseIt() throws Exception {
         when(localBinaryVersions.listExisting()).thenReturn(binaryVersions);
         assertThat(binaryVersionFetcher.fetchBinary(), equalTo(versionFullPath("8.8.8")));
     }
+
+    @Test
+    public void testWhenActiveVersionExistsItWillUseIt() throws Exception {
+        Preferences preferences = Preferences.userNodeForPackage(BootstrapperSupport.class);
+        preferences.put("bootstrapped version","8.8.8");
+        when(localBinaryVersions.activeVersion()).thenReturn(Optional.of(new BinaryVersion("7.8.9")));
```

Comment:
extract consts with names that show the intention behind them. 
E.g. 
* `"7.8.9"` -> `ROLLED_BACK_VERSION`
* `"8.8.8"` -> `SAVED_NEW_VERSION`